word_segmentation.py

```diff
@@ -0,0 +1,709 @@
+import numpy as np
+import os
+import icu
+import datetime
+from icu import UnicodeString, BreakIterator, Locale
+import matplotlib.pyplot as plt
+
+import tensorflow as tf
+from keras.models import Sequential
+from keras.layers import LSTM
+from keras.layers import Dense
+from keras.layers import TimeDistributed
+from keras.layers import Bidirectional
+from keras.layers import Embedding
+from keras.layers import Dropout
+
+
+def is_all_english(str):
```

Comment:
More formally, what you want to check is if the string is ASCII.  See some solutions here:

https://stackoverflow.com/q/196345/1407170